### DISCUSSION TITLE
Sorting of products in lunar panel

### DISCUSSION THREAD

**Discussion by yasirmturk** [lunarphp/lunar#2032] [Feature Requests]
Hi team, in the lunar panel for example there is no option to sort. 

It makes sense to sort the products based on created date DESC but it always shows ASC and admin has to next next many pages. 

I know search is there but it is using 3rd party search api in most cases which costs money. Another idea is to make admin search configurable using different service or database by default 🤔

**Comment by glennjacobs** [comment]
I think we need to consider relevancy of search for this. If we always order by a field, then finding a specific customer or w.e. becomes a lot harder.

**Reply by yasirmturk** [reply]
no, I meant allow sorting by some picked columns like name, created, things which can not be grouped otherwise. 
for example we can find product of a specific brand or collection in other views so we might not need sorting based on that

**Comment by dev3k** [comment]
I have a work around , in AppServiceProvider

```php
use Filament\Tables\Table;

 public function boot(): void
    {
        Table::configureUsing(function (Table $table): void {
            $table->defaultSort('id', 'desc');
        });
    }
```
any table without defaultSort will start sorting by id desc, I just started with lunar so I'm not sure if there some places might get affected, but it works for products

**Reply by dev3k** [reply]
maybe a safer approach 

```php
 public function boot(): void
    {
        Table::configureUsing(function (Table $table): void {
            $table->defaultSort(function ($query) {
                if ($query->getQuery()->from === 'products') {
                    return $query->orderBy($query->getQuery()->from.'.id', 'desc');
                }
            });
        });
    }
```
